server/server.py

```diff
@@ -14,11 +14,12 @@ def index():
 def hello():
     text=''
     lang=''
-    if request.method == "POST":
+    if request.method == "GET":
         text = request.form["text"]
         lang = request.form["lang"]
         print(text)
-    return speak.get_speak(text,lang)
+        speak.get_speak(text,lang)
+        return flask.url_for ("hello", filename="mp3_path")
```

Comment:
Changed it too, now it's working correctly and giving no errors.

Just one doubt, i have read the link you sent on the previous email about this and read the docs and i understand this line on speak.py and what it's for : 

`mp3_path = os.path.join(os.path.dirname(__file__), "tts.mp3")`

but if the file is already placed in the static folder what is this line for on server.py?

`return flask.url_for("hello", filename="mp3_path")`

It returns the file to the frontend ?

Reply:
Good question, `url_for('static', filename='filename.mp3')` will return a relative link to the generated file within the `static` folder which you've set to [/dist](https://github.com/CarlosAmorimProjects/TextToSpeech/blob/master/server/server.py#L8), so the front-end can access it. checkout this [doc](https://flask.palletsprojects.com/en/1.1.x/tutorial/static/). So basically it will return `/dist/filename.mp3` which the browser then will automatically resolve to `http://127.0.0.1:5000/dist/filename.mp3`